homeassistant/components/mqtt/discovery.py

```diff
@@ -62,6 +142,31 @@
             return
 
         payload = dict(payload)
+
+        for key in list(payload.keys()):
```

Comment:
Wait, how does this work, I was under the impression that we had a fixed list of abbreviation -> full key, which would not require a regex. If we can, let's avoid regex.

Reply:
It's a (long) fixed list of abbreviated _words_, the purpose of the regex is to split the configuration variable name in words and attempt to expand word by word.
Writing this comment, I'm not sure why I went for a regex though, a string.split() would have been more readable.